client/signup/config/steps-pure.js

```diff
@@ -383,7 +386,18 @@ export function generateSteps( {
 
 		'site-information': {
 			stepName: 'site-information',
-			providesDependencies: [ 'siteTitle', 'address', 'email', 'phone' ],
+			apiRequestFunction: createSiteWithCart,
+			providesDependencies: [
+				'siteTitle',
+				'address',
+				'email',
+				'phone',
+				'siteId',
+				'siteSlug',
+				'domainItem',
+				'themeItem',
+			],
+			delayApiRequestUntilComplete: true,
```

Comment:
Why do this?

Reply:
The way the signup flow is meant to work is that steps are only submitted when their dependencies are satisfied. If steps are being submitted to the API before their dependencies are being met then there must be a problem with the framework.

Reply:
>  If steps are being submitted to the API before their dependencies are being met

I thought the `delayApiRequestUntilComplete` should be there to prevent that problem, but the signup flow and the cart work well without the prop.

Reply:
So the prop it removed now.